File: problems/0009_palindrome-number/reversed-int/Solution.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0)
            return false;
        else if (x < 10)
            return true;
        int reversed = 0;
        for (int k = x; k != 0; k /= 10)
            reversed = reversed * 10 + (k % 10);
        return x == reversed;
    }
}

File: problems/0009_palindrome-number/reversed-int/Solution.kt
class Solution {
    fun isPalindrome(x: Int): Boolean {
        if (x < 0)
            return false
        else if (x < 10)
            return true
        var reversed = 0
        var k = x
        while (k != 0) {
            reversed = reversed * 10 + (k % 10)
            k /= 10
        }
        return x == reversed
    }
}
